android/app/src/main/java/com/on/turip/data/common/SafeApiCall.kt

```diff
@@ -1,31 +1,48 @@
 package com.on.turip.data.common
 
+import kotlinx.coroutines.CancellationException
+import kotlinx.serialization.json.Json
 import retrofit2.HttpException
 import retrofit2.Response
+import java.io.IOException
 
-@Suppress("UNCHECKED_CAST")
-suspend inline fun <T> safeApiCall(apiCall: suspend () -> Response<T>): TuripCustomResult<T> =
+suspend fun <T> safeApiCall(apiCall: suspend () -> Response<T>): TuripCustomResult<T> {
```

Comment:
```suggestion
suspend inline fun <T> safeApiCall(apiCall: suspend () -> Response<T>): TuripCustomResult<T> {
```
고차함수라 다음과 같이 inline을 붙이는 게 더 좋을 거 같아요!

Reply:
고차함수긴 한데 suspend 함수고 해당 함수는 인라이닝이 성능상 이점이 없다고 하네요

inline 키워드를 붙이면 apiCall 부분의 suspend 에 
`Redundant 'suspend' modifier: lambda parameters of suspend function type uses existing continuation.`
문구가 나오고 제안하는 행동을 따라가면  apiCall 부분의 suspend를 제거하도록 되고 safeApiCall의 suspend도 사용되지 않는 상태로 표시되어서 Continuation 객체를 사용할 수 없지 않나요..?

최종적으로 inline 키워드를 붙이면 
```kotlin
inline fun <T> safeApiCall(apiCall: () -> Response<T>): TuripResult<T> {
```
이 형태로 사용하라는 것 같아요